cli/ts/commands/genProofs.ts

```diff
@@ -240,10 +240,19 @@ export const genProofs = async ({
     }
   } else {
     // build an off-chain representation of the MACI contract using data in the contract storage
-    let fromBlock = startBlock ? Number(startBlock) : 0;
+    const defaultStartBlock = await maciContract
+      .queryFilter(maciContract.filters.SignUp())
```

Comment:
I think the problem with this approach is that if someone deploys a poll before the first signup is made, then any log before the first signup will be lost, like the poll deployment event.
My suggestion is that for the cli we can put it as mandatory param, and for the contract task we save (same way as we save all other data about the deployments) the Maci deployment block and use it later

Reply:
So, I added poll event parsing as well so if poll is deployed before signup, it takes poll event block and vice versa.